package/upgrade/upgrade_manifests.sh

```diff
@@ -1299,6 +1299,80 @@ wait_for_deployment() {
   done
 }
 
+upgrade_rbac() {
+
+  # only versions before v1.3.0 that upgrading to v1.3.0 need this patch
+  if [[ ! "${UPGRADE_PREVIOUS_VERSION%%-rc*}" < "v1.2.2" ]]; then
+    echo "Only versions before v1.2.2 need this patch."
+    return
+  fi
+
+  full_names=$(upgrade-helper get-cloud-provider)
+  ret=$?
+  if [ -z "$full_names" ] || [ $ret != 0 ]; then
+    echo "No rolebinding harvesterhci.io:cloudprovider found or command failed, we can skip to patch rbac this moment."
+    return
+  fi
+
+  clusterrole_manifest="/tmp/clusterrole.yaml"
+  clusterrolebinding_manifest_prefix="clusterrolebinding"
+
+  # clean up the tmp files if they exist
+  rm -f "$clusterrole_manifest"
+  rm -f "/tmp/${clusterrolebinding_manifest_prefix}*.yaml"
+
+  cat > "$clusterrole_manifest" <<EOF
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  labels:
+    app.kubernetes.io/component: apiserver
+    app.kubernetes.io/name: harvester
+    app.kubernetes.io/part-of: harvester
+  name: harvesterhci.io:csi-driver
+rules:
+- apiGroups:
+  - storage.k8s.io
+  resources:
+  - storageclasses
+  verbs:
+  - get
+  - list
+  - watch
+EOF
+
+  echo "Creating clusterrole with ${clusterrole_manifest} ..."
+  kubectl create -f "$clusterrole_manifest"
```

Comment:
Seeing the error when testing with https://github.com/harvester/harvester/pull/5558:
```
Creating clusterrole with /tmp/clusterrole.yaml ...
Error from server (AlreadyExists): error when creating "/tmp/clusterrole.yaml": clusterroles.rbac.authorization.k8s.io "harvesterhci.io:csi-driver" already exists
```

With #5558, the harvester pod creates the cluster role after the harvester pod is upgraded.

Reply:
Good catch... the upgrade will apply the new template of the chart

Add another checking for this op.
Please help to recheck, thanks!